tests/kafkatest/tests/core/quorum_reconfiguration_test.py

```diff
@@ -46,54 +50,67 @@ def setUp(self):
+                                                        broker_only_ids), timeout_sec=5)
         # Start second controller
         self.kafka.controller_quorum.add_broker(inactive_controller)
-        output = self.kafka.describe_quorum()
-        assert re.search(r"LeaderId:\s*" + str(active_controller_id), output)
-        assert_nodes_in_output(r"CurrentVoters:.*", output, active_controller_id)
-        assert_nodes_in_output(r"CurrentObservers:.*", output, *broker_ids + [inactive_controller_id])
-
+        wait_until(lambda: check_describe_quorum_output(self.kafka.describe_quorum(),
+                                                        active_controller_id,
+                                                        [active_controller_id],
+                                                        broker_only_ids + [inactive_controller_id]), timeout_sec=5)
         # Add controller to quorum
         self.kafka.controller_quorum.add_controller(inactive_controller_id, inactive_controller)
 
         # Check describe quorum output shows both controllers are voters
-        output = self.kafka.describe_quorum()
-        assert re.search(r"LeaderId:\s*" + str(active_controller_id), output)
-        assert_nodes_in_output(r"CurrentVoters:.*", output, active_controller_id, inactive_controller_id)
-        assert_nodes_in_output(r"CurrentObservers:.*", output, *broker_ids)
-
+        wait_until(lambda: check_describe_quorum_output(self.kafka.describe_quorum(),
+                                                        active_controller_id,
+                                                        [active_controller_id, inactive_controller_id],
+                                                        broker_only_ids), timeout_sec=5)
         # Remove leader from quorum
-        voters = json_from_line(r"CurrentVoters:.*", output)
+        voters = json_from_line(r"CurrentVoters:.*", self.kafka.describe_quorum())
         directory_id = next(voter["directoryId"] for voter in voters if voter["id"] == active_controller_id)
         self.kafka.controller_quorum.remove_controller(active_controller_id, directory_id)
-
-        # Check describe quorum output to show second_controller is now the leader
-        output = self.kafka.describe_quorum()
-        assert re.search(r"LeaderId:\s*" + str(inactive_controller_id), output)
-        assert_nodes_in_output(r"CurrentVoters:.*", output, inactive_controller_id)
-        assert_nodes_in_output(r"CurrentObservers:.*", output, *broker_ids)
-
+        # Check describe quorum output to show second_controller is now the leader. Observers contain the old controller
+        wait_until(lambda: check_describe_quorum_output(self.kafka.describe_quorum(),
+                                                        inactive_controller_id,
+                                                        [inactive_controller_id],
+                                                        broker_only_ids), timeout_sec=5)
```

Comment:
The reason why the leader is not in the observer is related to my other comment. `check_describe_quorum_output` always returns true.

Reply:
we discussed offline that technically wait_until should work on asserts. we found the reason the old leader was not in the observers is due to KAFKA-17608